source/game.c

```diff
@@ -2156,11 +2157,27 @@ static void played_cards_update_loop(bool* discarded_card, int* played_selection
                     }
                     break;
                 
+                case PLAY_BEFORE_SCORING:
+
+                    // Activate Jokers with an effect just before the hand is scored
+                    if (check_and_score_joker_for_event(&joker_scored_index, NULL, JOKER_EVENT_ON_HAND_PLAYED))
+                    {
+                        return;
+                    }
+
+                    joker_scored_index = 0;
+                    play_state = PLAY_SCORING_CARDS;
+                    
+                    if (card_object_is_selected(played[i]) && played_top - i >= *played_selections)
+                    {
+                        played_y -= int2fx(10);
+                    }
+                    break;
```

Comment:
```suggestion
                    
                    if (card_object_is_selected(played[i]) && played_top - i >= *played_selections)
                    {
                        played_y -= int2fx(10);
                    }
                     
                    joker_scored_index = 0;
                    play_state = PLAY_SCORING_CARDS;
                    break;
```

Move the state transition to the end